phoenix-core/src/main/java/org/apache/phoenix/query/ConnectionQueryServicesImpl.java

```diff
@@ -916,6 +916,9 @@ private void addCoprocessors(byte[] tableName, TableDescriptorBuilder builder, P
                 } else if (indexRegionObserverEnabled && !newDesc.hasCoprocessor(GlobalIndexChecker.class.getName()) &&
                         !isLocalIndexTable(newDesc.getColumnFamilyNames())) {
                     builder.addCoprocessor(GlobalIndexChecker.class.getName(), null, priority - 1, null);
+                    if(newDesc.hasCoprocessor(IndexRegionObserver.class.getName())) {
+                        builder.removeCoprocessor(IndexRegionObserver.class.getName());
```

Comment:
@kadirozde this was added to remove the data table coprocessor that gets added on an index table (it is similar to having indexer coproc (as per old design) added to all the tables by default). Shall we do this?

Reply:
Ok